### ISSUE TITLE
[git] Support Git `push`, `pull`, `fetch`, and `merge` commands with arguments

### ISSUE THREAD

**Issue by kittaakos** [theia-ide/theia#583]
Pull, push and fetch should list the remotes.
Merge should list the list of known remote and local branches.

**Comment by svenefftinge** [comment]
In vscode they use the quick-command to select the origin on push/fetch/pull.

**Comment by svenefftinge** [comment]
The good thing about it, is that it allows to execute the command from anywhere.

**Comment by marcdumais-work** [comment]
I am trying the git extension using my Theia development environment. I have a few remotes defined. ATM I see that for pull and push operations, a random remote is picked (1st returned by "git remote" ?). It happens to be one I do not have permissions to push-to, and that doesn't contain the branch I am working on,  so push and pull operations fail. Fetch doesn't seem to do anything and outputs no error message.

I guess these problems will be addressed in this issue by permitting to pick a specific remote for the operations?

push:
``` 
The command `git push antoine task` exited with an unexpected code: 1. The caller should either handle this error, or expect that exit code.
To github.com:hexa00/theia.git
 ! [remote rejected]   task -> task (permission denied)
error: failed to push some refs to 'git@github.com:hexa00/theia.git'
```
pull: 
```
The command `git pull antoine` exited with an unexpected code: 1. The caller should either handle this error, or expect that exit code.
You asked to pull from the remote 'antoine', but did not specify
a branch. Because this is not the default configured remote
for your current branch, you must specify a branch on the command line.
```

**Comment by kittaakos** [comment]
> I am trying the git extension using my Theia development environment.

@marcdumais-work, I have moved this to a separate task: https://github.com/theia-ide/theia/issues/621

**Comment by kittaakos** [comment]
Will be fixed as part of #621.
